src/main/scala/scala/tools/partest/nest/StreamCapture.scala

```diff
@@ -50,4 +51,42 @@ object StreamCapture {
     }
     Captured(stdoutFn(), stderrFn(), result)
   }
+
+  def withExtraProperties[A](extra: Map[String, String])(action: => A): A = {
+    val saved = System.getProperties()
+    val modified = new java.util.Properties()
+    saved.stringPropertyNames().forEach((k) => modified.setProperty(k, saved.getProperty(k)))
```

Comment:
```
val modified = saved.clone.asInstanceOf[java.util.Properties]
modified.putAll(extra.asJava)
```  
?

Reply:
I've settled on `modified.putAll(saved)` to avoid the cast after cloning.